data/src/main/java/app/otakureader/data/worker/ReadingReminderWorker.kt

```
@@ -0,0 +1,128 @@
+
+            val dailyGoal = readingGoalPreferences.dailyChapterGoal.first()
+            val zone = ZoneId.systemDefault()
+            val startOfDayMs = LocalDate.now(zone).atStartOfDay(zone).toInstant().toEpochMilli()
+            val chaptersToday = readingHistoryDao.getChaptersReadSince(startOfDayMs).first()
+
+            // Don't remind if daily goal is set and already met
+            if (dailyGoal > 0 && chaptersToday >= dailyGoal) return Result.success()
+
+            showNotification(dailyGoal, chaptersToday)
+            Result.success()
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)